keps/420-partial-admission/README.md

```diff
@@ -0,0 +1,148 @@
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Add and optional way of allowing the partial admission of a workload if the full admission is not possible.
+
+## Motivation
+
+In practice, not all Workloads require the parallel execution of all the `count` of a `PodSet`, for such cases having a way to partially reserve the quota in order to prevent starvation.
+
+For example if a batch/Job has parallelism is x and there is only quota available for y < x then jobs should still be admitted.
+
+### Goals
+
+Provide an opt-in way for Workloads to accept the admission with a lower count of pods if the full count is not available.
+
+### Non-Goals
+
+Since this is an opt-in feature, the parent job should accept the partial admission parameters provided by Kueue.
+
+Kueue will not take any measure to ensure that the parent job respects the assigned quota.
+
+## Proposal
+
+Change the way the flavor assigner work to support decrementing the pods count in order to find a better fit for the current workload.
+In case a partial fit is chosen, the jobframework reconciler should provide the admitted pod counts to the parent job before unsuspending it, in a similar fashion as the node selectors are provided. In case the job gets suspended, the original pod counts should be restored in order to allow a potential future admission with its original pod counts.
+
+### User Stories
+
+Kueue issue [420](https://github.com/kubernetes-sigs/kueue/issues/420) provides details on the initial feature details and its applicability for `batch/Job`.
+
+### Risks and Mitigations
+
+TBD: It should be decided if the partial admission should be taken into account before attempting preemption. Since preemption has its own penalties (time to evict the active workloads, potential waste of the resources used by the preempted workloads) I vote to take it into account before.
+
+## Design Details
+
+### Workload API
+
+```go
+type PodSet struct {
+    // .......
+
+    // count is the number of pods for the spec.
+    // +kubebuilder:validation:Minimum=1
+    Count int32 `json:"count"`
+
+    // minimumCount is the minimum number of pods for the spec acceptable
+    // in case of partial admission.
+    //
+    // If not provided, partial admission for the current PodSet is not
+    // enabled.
+    // +optional
+    MinimumCount *int32 `json:"minimumCount,omitempty"`
+}
+
+```
+
+### Scheduler / Flavorassignment
+
+I case the current workload dose not fit in the current available quota and any of its PodSets allow partial admission and the count did not reach the `MinimumCount`, try to find a combination of counts that fits. The accepted number of pods in each PodSet are published in `workload.Status.Admission.PodSetAssignments[*].ResourceUsage.Count`
```

Comment:
Can you provide a high level summary of how you plan to implement "try a combination of counts that fits"?

I know you have a working PR, but it might be helpful to lay down some high level idea here, both to being able to spot design flaws and to have a future reference.

Here is an example of the level of detail that I'm looking for https://github.com/kubernetes-sigs/kueue/tree/main/keps/83-workload-preemption#changes-in-scheduling-algorithm
It might be simpler for this problem.

Reply:
done